events/api/src/main/java/org/projectnessie/events/api/ReferenceType.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.events.api;
+
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+/** An enum of all possible reference types. */
+@Schema(
+    description = "An enum of all possible reference types.",
+    enumeration = {"BRANCH", "TAG"})
+public enum ReferenceType {
```

Comment:
Probably worth to add the same "fallback" logic as introduced in #6634

Reply:
Do we really need that though? Nessie does not need to deserialize this from json. Only users will. I'd say it's their business :-)

Reply:
If we do not serialize it but expose internal names, should we have a mapping from `ReferenceEvent.getFullReferenceName()` to `ReferenceType`?

Reply:
I agree that it's a "user concern" similar to Git, where the "simple name mapping" is rather a tool's thing.
But not opposed to have convenience methods (not "real" attributes) on the data types.

Reply:
Not quite, I'm suggesting to have something like `isBranch()` derived from `getFullReferenceName()` (not sure it is possible, though)

Reply:
Ah OK I can add that too, but it will likely be derived from `getReferenceType()`.

Reply:
Now, I'm confused :) I thought we were going to drop `ReferenceType` from JSON, not so?

Reply:
> Do we really need that though?

I was referring to the "fallback logic mentioned by @snazy :-)

Reply:
Well, I think if we add `@JsonSerialize` we have to support schema/sub-type evolution too. Clients will probably be able to use a lenient attribute parsing mode, but will they be able to handle new sub-types if we do have a dedicated serializer? I guess the serializer has to be defined in our code. Not sure if clients can add that on top of our jars. WDYT?

Reply:
Yes, you are both right. In the case where a newer event type is created and sent by Nessie, a consumer that has an older version of the Events API will not be able to deserialize it.

I will rework this.

Reply:
Done in 66a19a2.

Reply:
I pushed one more commit and now I think everything is taken care of. The 3 enums in this package were handled as follows:

1. `EventType` is handled by a special (de)serializer that deserializes unknown events as `CustomEvent`;
2. `ContentType` is handled by a special (de)serializer that deserializes unknown contents as `CustomContent`; 
3. `ReferenceType` was deleted completely. I was convinced by @dimas-b :-)

Let me know what you think.